netket/experimental/operator/_fermion_operator_2nd.py

```diff
@@ -236,11 +243,52 @@ def __repr__(self):
             f"n_operators={len(self._operators)}, dtype={self.dtype})"
         )
 
+    def reduce(self, order: bool = True, inplace: bool = True):
+        """Prunes the operator by removing all terms with zero weights, grouping, and normal ordering (inplace)."""
+        operators = self._operators
+        terms, weights = list(operators.keys()), list(operators.values())
+        if order:
+            terms, weights = _normal_ordering(terms, weights)
+        terms, weights, constant = _collect_constants(terms, weights)
+        operators = dict(zip(terms, weights))
+        operators = _reduce_operators(operators, self.dtype)
+        constant = self._constant + constant
+        if inplace:
+            self._operators = operators
+            self._constant = constant
+        else:
+            new = FermionOperator2nd(
+                self.hilbert, constant=self._constant + constant, dtype=self.dtype
+            )
+            new._operators = operators
+            new._constant = constant
+            return new
```

Comment:
I would make this only inplace, or is there some reason to have the non-inplace version? If one wants the non inlace one, he can just `.copy().reduce()`